### ISSUE TITLE
firebase_auth: <iOS after update to latest version phone auth returns invalid-app-credentials>

### ISSUE THREAD

**Issue by tdujmic** [firebase/flutterfire#13284]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Which plugins are affected?

Core, Auth

### Which platforms are affected?

iOS

### Description

After manage to initialize firebase app in main of my existing app and updated plugins to latest versions under flutter 3.24.1
I tried to send a normal verifyphonenumber request but I receive always error code invalid-app-credentials ?? 

### Reproducing the issue

    FirebaseApp app = await Firebase.initializeApp(
      options: DefaultFirebaseOptions.currentPlatform,
    );
    FirebaseAuth _auth = FirebaseAuth.instanceFor(app: app);

    await _auth.verifyPhoneNumber(
      phoneNumber: parsedNumber.toString(),
      verificationCompleted: (PhoneAuthCredential credential) async {
        print('verification completed '+ credential.toString());
      },
      verificationFailed: (FirebaseAuthException e){
        print('verification error '+ e.code);
      },
      codeSent: (String verificationId, int? resendToken) async {
              print(verificationId.toString());

      },
      codeAutoRetrievalTimeout: (String verificationId) {
        print('Verification id ' + verificationId.toString());
      },

    );

### Firebase Core version

3.4.0

### Flutter Version

3.24.1

### Flutter dependencies

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml

Replace this line with the contents of your `flutter pub deps -- --style=compact`.

```

</details>

**Comment by ignacioss** [comment]
I have the same issue

**Comment by SelaseKay** [comment]
Hi @tdujmic , thanks for the report. Are you experiencing this on an actual device or a simulator?

**Comment by panicoli0** [comment]
We experienced the same issue, but it started working 2 days after setup.

**Comment by tdujmic** [comment]
@SelaseKay yes I experienced that on my phisical iPgone 15 Pro Max and my coueague iPhone 11 Pro means its something with parameters providing by plugin to firebase servers, or we need maybe to take some more actions with removal of old type of integration used on flutter versions 3.16 or 3.22, I have checked our firebase account plan and is upgrades from spark since firebase phone auth is not free from 1st of September. But this doesn't help. Have to try lot of times to dovnload and refresh Google Service info plist file, start flutterfire reconfigure recheck params in info plist but no use. 
Couple of minutes ago have to test it last time and still get the same error message. App is initialised properly, all ID's presented in firebase app options dump before I send validatephone request.

If you need soem more info just post the message or contact me over email, Im available for some workshop session too if is needed. Just let me know please.

Thanks in Advance

**Comment by tdujmic** [comment]
@SelaseKay there you are. Its a part of my pubsec.yaml :


  cloud_firestore: ^5.4.0
  firebase_storage: ^12.2.0
  firebase_auth: ^5.2.0
  firebase_core: ^3.4.0

  firebase_messaging: ^15.1.0
  firebase_database: ^11.1.1
  firebase_analytics: ^11.3.0

  firebase_core_platform_interface: ^5.2.1

**Comment by SelaseKay** [comment]
Kindly try the solutions described [here](https://stackoverflow.com/questions/74340881/invalid-app-credential-token-mismatch-in-flutter-firebase-auth). Please let me know if it was of help.

**Comment by heshesh2010** [comment]
@SelaseKay hello , 

I have tested the solutions and still got same error [firebase_auth]: IOS real device invalid-app-credential #13330

**Comment by AyeshaIftikhar** [comment]
> Kindly try the solutions described [here](https://stackoverflow.com/questions/74340881/invalid-app-credential-token-mismatch-in-flutter-firebase-auth). Please let me know if it was of help.

Did not work for me

**Comment by SelaseKay** [comment]
There's a PR linked to this issue. That should resolve it when it's merged.
